packages/@webex/plugin-logger/src/logger.js

```diff
@@ -415,9 +423,18 @@ function makeLoggerMethod(level, impl, type, neverPrint = false, alwaysBuffer =
 
         stringified.unshift(logDate.toISOString());
         stringified.unshift('|  '.repeat(this.groupLevel));
-        buffer.push(stringified);
-        if (buffer.length > historyLength) {
-          buffer.splice(0, buffer.length - historyLength);
+        bufferRef.buffer.push(stringified);
+        if (bufferRef.buffer.length > historyLength) {
+          // we've gone over the buffer limit, trim it down
+          const deleteCount = bufferRef.buffer.length - historyLength;
+
+          bufferRef.buffer.splice(0, deleteCount);
```

Comment:
when incremental logs config is enabled we could potentially upload them here instead of just deleting old ones when history limit is reached, but this is out of scope for this PR, I've raised https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-610060 for this